openxr/src/lib.rs

```diff
@@ -82,8 +82,16 @@ fn fixed_str_bytes(x: &[c_char]) -> &[u8] {
 
 fn get_str(mut getter: impl FnMut(u32, &mut u32, *mut c_char) -> sys::Result) -> Result<String> {
     let mut bytes = get_arr(|x, y, z| getter(x, y, z as _))?;
-    // Strip null byte
-    bytes.truncate(bytes.len() - 1);
+    // Terminate at first null byte
+    let mut first_nt = bytes.len();
+    for (index, byte) in bytes.iter().enumerate() {
+        if *byte == 0 && index <= first_nt {
+            first_nt = index;
+        }
+    }
```

Comment:
```suggestion
    let mut first_nt = bytes.iter().rfind(|&&x| x != 0).expect("missing null") + 1;
```

Reply:
Since it's Double Ended Iterator this would return the actual byte instead of the index. Additionally, assuming index was returned,  this would break after the first occurrence from the rear leaving the last one in place. I've updated my implementation to use `position` which have confirmed to work, but is still satisfyingly terse.

Reply:
Whoops, right, should be `rposition(|&x| ...)`